File: archive/latest-time-by-replacing-hidden-digits.js
const maximumTime = function(time) {
  time = time.split('')

  if (time[0] === '?') {
    if (time[1]!=="?") {
      if (Number(time[1]) > 3) {
        time[0] = '1'
      } else {
        time[0] = '2'
      }
    } else {
      time[0] = '2'
    }
  }
  if (time[1] === '?') {
    if (time[0] === '0') {
      time[1] = '9'
    }
    if (time[0] === '1') {
      time[1] = '9'
    }
    if (time[0] === '2') {
      time[1] = '3'
    }
  }
  if (time[3] === '?') {
    time[3] = '5'
  }
  if (time[4] === '?') {
    time[4] = '9'
  }

  return time.join('')
}

console.log(maximumTime("2?:?0"))

File: go-archive/latest-time-by-replacing-hidden-digits.go
func maximumTime(time string) string {
    ans:=""
    for i, v := range time {
        if (v != '?') {
            ans+=string(v)
            continue
        }

        if i == 0 {
            if (
                time[1] == '0' ||
                time[1] == '1' ||
                time[1] == '2' ||
                time[1] == '3' ||
                time[1] == '?') {
                ans+="2"
            } else {ans+="1"}
        }

        if i == 1 {
            if (ans[0] == '1' || ans[0] == '0') {
                ans+="9"
            } else {ans+="3"}
        }

        if i == 3 {ans+="5"}
        if i == 4 {ans+="9"}
    }

    return ans
}